Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs

```diff
@@ -0,0 +1,244 @@
+            }
+        }
+
+        /// <summary>
+        /// Call to request that the scanning progress be finishing. The application must do
+        /// this to finalize the playspace. The scan state will not progress pass
+        /// Scanning until this is called. The spatial understanding queries will not function
+        /// until the playspace is finalized.
+        /// </summary>
+        public void RequestFinishScan()
+        {
+            if (SpatialUnderstanding.Instance.AllowSpatialUnderstanding)
+            {
+                SpatialUnderstandingDll.Imports.GeneratePlayspace_RequestFinish();
+                ScanState = ScanStates.Finishing;
+            }
+        }
+
+        /// <summary>
+        /// Update the scan progress. This function will initialize the scan, update it, 
+        /// and issue a final mesh import, when the scan is complete.
+        /// </summary>
+        /// <param name="deltaTime">The amount of time that has passed since the last update (typically Time.deltaTime)</param>
+        private void Update_Scan(float deltaTime)
+        {
+            // If we auto-start scanning, do it now
+            if (AutoBeginScanning &&
+                (ScanState == ScanStates.None))
+            {
+                RequestBeginScanning();
+            }
+
+            // Update the scan
+            bool scanDone = false;
+            if (((ScanState == ScanStates.ReadyToScan) ||
+                 (ScanState == ScanStates.Scanning) ||
+                 (ScanState == ScanStates.Finishing)) &&
+                (SpatialUnderstanding.Instance.AllowSpatialUnderstanding))
+            {
+                // Camera
+                Vector3 camPos = Camera.main.transform.position;
+                Vector3 camFwd = Camera.main.transform.forward;
+                Vector3 camUp = Camera.main.transform.up;
+
+                // If not yet initialized, do that now
+                if (ScanState == ScanStates.ReadyToScan)
+                {
+                    SpatialUnderstandingDll.Imports.GeneratePlayspace_InitScan(
+                        camPos.x, camPos.y, camPos.z,
+                        camFwd.x, camFwd.y, camFwd.z,
+                        camUp.x, camUp.y, camUp.z,
+                        ScanSearchDistance, ScanSearchDistance);
+                    ScanState = ScanStates.Scanning;
+                }
+
+                // Update
+                int meshCount;
+                IntPtr meshList;
+                if (UnderstandingSourceMesh.GetInputMeshList(out meshCount, out meshList))
+                {
+                    scanDone = SpatialUnderstandingDll.Imports.GeneratePlayspace_UpdateScan(
+                        meshCount, meshList,
+                        camPos.x, camPos.y, camPos.z,
+                        camFwd.x, camFwd.y, camFwd.z,
+                        camUp.x, camUp.y, camUp.z,
+                        deltaTime) == 1;
+                }
+            }
```

Comment:
I don't know if this is intentional or desired, but it looks like GeneratePlayspace_UpdateScan will get called even for the ScanStates.Finishing state. It may be desirable to grab one more final update, or it may be desirable to skip another update since the finalization process seems quite intensive. Thought I would at least point it out.